File: coodingpenguin/leetcode/217_Contains_Duplicate.py
# 문제: 217. Contains Duplicate
# 링크: https://leetcode.com/problems/contains-duplicate/

# 시간/공간: O(N) / O(N)
class Solution:
    def containsDuplicate(self, nums: List[int]) -> bool:
        return len(nums) > len(set(nums))        

File: sally/leetcode/217_contains-duplicate.cpp
// 217. Contains Duplicate (Easy)
// Runtime: 16 ms, faster than 97.40% of C++ online submissions for Contains Duplicate.
// Memory Usage: 15.5 MB, less than 84.13% of C++ online submissions for Contains Duplicate.

class Solution {
public:
    bool containsDuplicate(vector<int>& nums) {
        sort(nums.begin(), nums.end());
        int N = nums.size();
        if(N == 1) return false;
        for(int i = 1; i < N; i++)
            if(nums[i-1] == nums[i]) return true;
        return false;
    }
};